tests/bsim/bluetooth/mesh/src/mesh_test.c

```diff
@@ -546,6 +546,46 @@ uint16_t bt_mesh_test_own_addr_get(uint16_t start_addr)
 	return start_addr + get_device_nbr();
 }
 
+void bt_mesh_test_send_over_adv(void *data, size_t len)
+{
+	struct bt_mesh_adv *adv = bt_mesh_adv_create(BT_MESH_ADV_DATA, BT_MESH_ADV_TAG_LOCAL,
+						     BT_MESH_TRANSMIT(0, 20), K_NO_WAIT);
+	net_buf_simple_add_mem(&adv->b, data, len);
+	bt_mesh_adv_send(adv, NULL, NULL);
+}
+
+int bt_mesh_test_wait_for_adv(struct k_sem *observer_sem, bt_le_scan_cb_t scan_cb)
```

Comment:
It waits for any frame, not only adv. Scanner with these settings can receive gatt frames too.

Reply:
Will look into something similar to beacon test.

Reply:
As I wrote above, the function name is not correct. It receives not only over advertising but over gatt too.
It should be something like: `bt_mesh_test_wait_for_packet` or `bt_mesh_test_wait_for_frame`.

Reply:
Changed to `bt_mesh_test_wait_for_packet`